### PR TITLE
Improve coverage in mvn verify check github workflow

### PR DESCRIPTION
**Pull request by gerashegalov**
Addresses the issue slipped into #6141 . Workflow improvements:
- Dynamically generate builds for all shims without hardcoding versions using https://docs.github.com/en/actions/learn-github-actions/expressions#fromjson
- Imitate `buildall`. Verify shim-specific code is buildable in parallel. In another parallel check RAT, scalastyle, and verify all modules with the default spark version
- add default value `false` for maven.scalastyle.skip 
    
Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

File: .github/workflows/mvn-verify-check.yml [hunk 1]
```diff
@@ -24,7 +24,32 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  mvn-verify-check:
+  package-aggregator:
+    strategy:
+      matrix:
+        spark-version: [312, 313, 320, 321, 322, 330]
```
Comment (tgravescs) [inline .github/workflows/mvn-verify-check.yml:30]: we have yet another place to update versions?  What is this list supposed to match?  why doesn't it have 321cdh?

Comment (gerashegalov) [reply .github/workflows/mvn-verify-check.yml:30]: > we have yet another place to update versions? 

Unfortunately yes. There might be an elegant solution to generate this array in an upstream job, but I haven't looked at it yet. I would really like us to solve this once and for all including CI. 

> What is this list supposed to match? why doesn't it have 321cdh?

it's supposed to be like noSnapshots. I first had 321cdh too but then decided against a vendor shim in a github. I'll put it back for consistency. I don't have a particularly strong point for not including it.

Comment (tgravescs) [reply .github/workflows/mvn-verify-check.yml:30]: all the jars are available open source so I think should include it like normal build does

**Comment by gerashegalov** [comment]
build

File: .github/workflows/mvn-verify-check.yml [hunk 2]
```diff
@@ -24,7 +24,61 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  mvn-verify-check:
+  get-noSnapshot-versions-from-dist:
+    runs-on: ubuntu-latest
+    outputs:
+      sparkHeadVersion: ${{ steps.noSnapshotVersionsStep.outputs.headVersion }}
```
Comment (pxLi) [inline .github/workflows/mvn-verify-check.yml:30]: looks like we are using this headVersion(311) here? for `verify-all-modules-with-headSparkVersion` we use default buildver

Comment (gerashegalov) [reply .github/workflows/mvn-verify-check.yml:30]: correct, I ended not needing it. But actually it makes the flow more clear if I use it explicitly. I'll update

**Comment by gerashegalov** [comment]
build

**Review by pxLi — APPROVED** [review]
neat! LGTM

**Comment by abellina** [comment]
re-triggering due to markdown link check failure.

**Review by revans2 — APPROVED** [review]
This looks good to me, but I am not expert and I am relying on the review from @pxLi for the workflow changes.

**Review by abellina — APPROVED** [review]

**Comment by gerashegalov** [comment]
> re-triggering due to markdown link check failure.
> 
> Note to self: just typing the `build` command didn't actually help, I went into the failing mardown workflow and re-triggered it there.

Only https://github.com/NVIDIA/spark-rapids/blob/branch-22.08/.github/workflows/blossom-ci.yml#L71 is triggerred by comment

**Pull request merged**